Content.Shared/Item/MakeItemOnDeadSystem.cs

```diff
@@ -0,0 +1,31 @@
+using Content.Shared.Mobs;
+using Content.Shared.Mobs.Components;
+
+namespace Content.Shared.Item
+{
+    /// <summary>
+    ///     Adds ItemComponent to entity when it dies. Remove when it revives.
+    /// </summary>
+    public sealed class MakeItemOnDeadSystem : EntitySystem
+    {
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<MakeItemOnDeadComponent, MobStateChangedEvent>(OnMobStateChanged);
+        }
+
+        private void OnMobStateChanged(EntityUid uid, MakeItemOnDeadComponent component, MobStateChangedEvent args)
+        {
+            TryComp<MobStateComponent>(uid, out var mobState);
+
+            if (mobState.CurrentState == MobState.Dead)
```

Comment:
There's 0 reason this can't use compregistry.

Reply:
im not sure what do you mean